src/providers/Dao.js

```diff
@@ -0,0 +1,51 @@
+import React, { useContext } from 'react'
+import { addressesEqual } from '@1hive/1hive-ui'
+import { useLocation } from 'react-router-dom'
+import daoList from '@1hive/gardens-dao-list'
+
+import { useWallet } from '../providers/Wallet'
+
+const DAOContext = React.createContext()
+
+export function parsePath(pathname, search = '') {
+  const [, ...parts] = pathname.split('/')
+
+  return parts[0]
+}
+
+function getConnectedAccountBalance(account, dao) {
+  return 0
+}
+
+export function DAOProvider({ children }) {
+  const location = useLocation()
+  const { account } = useWallet()
+
+  // get all the daos for the dashboard
+
+  // note that i am calling this daoId because at some point we might want to also handle aragon id's and not just addresses
+  const firstParameter = parsePath(location.pathname)
+
+  const connectedDAO = daoList.daos.find(d =>
+    addressesEqual(firstParameter, d.address)
+  )
+
+  // somehow here we are going to get the dao token info and the user balance for the dao
+  const accountBalance = getConnectedAccountBalance(account, connectedDAO)
+  const stakeToken = {}
+
+  const DAOInfo = {
+    gardensList: daoList.daos,
```

Comment:
Lets remove this as i don't think it will be needed here

Reply:
i mean, maybe the struct does not have the right naming, but what we are returning here is the list of daos (that we should take it from the subgraph and append the dao metadata from the daoList library) and also the connected dao if there is any

Reply:
Ah yeah sounds good. Maybe we could rename to `daoList` to keep it consistent